apps/web/src/components/purchase-form/cards/card-form.tsx

```diff
@@ -0,0 +1,101 @@
+import { CheckoutMethod, CheckoutStatus, PackType } from '@algomart/schemas'
+import clsx from 'clsx'
+import { useRouter } from 'next/router'
+import useTranslation from 'next-translate/useTranslation'
+import { FormEvent, useCallback, useState } from 'react'
+
+import CardPurchaseError from './sections/card-error'
+import CardPurchaseForm from './sections/card-form'
+import CardPurchaseHeader from './sections/card-header'
+import CardPurchaseSuccess from './sections/card-success'
+import CardPurchaseSummary from './sections/card-summary'
+
+import css from './card-form.module.css'
+
+import Loading from '@/components/loading/loading'
+import { PaymentContextProps } from '@/contexts/payment-context'
+import { useWarningOnExit } from '@/hooks/use-warning-on-exit'
+import { isAfterNow } from '@/utils/date-time'
+
+export default function CardForm({
+  bid,
+  release,
+  formErrors,
+  handleRetry,
+  handleSubmitBid: onSubmitBid,
+  handleSubmitPurchase: onSubmitPurchase,
+  loadingText,
+  packId,
+  price,
+  setBid,
+  status,
+}: PaymentContextProps) {
+  const { t } = useTranslation()
+  const { asPath, push } = useRouter()
+  const [promptLeaving, setPromptLeaving] = useState(false)
+  const isAuctionActive =
+    release?.type === PackType.Auction &&
+    isAfterNow(new Date(release.auctionUntil as string))
+  useWarningOnExit(promptLeaving, t('common:statuses.processingPayment'))
+
+  const handleSubmitPurchase = useCallback(
+    async (event: FormEvent<HTMLFormElement>) => {
+      event.preventDefault()
+      setPromptLeaving(true)
```

Comment:
Setting `setPromptLeaving` causes the app to throw a message confirming changing routes when there's a form error. This is because now we'll be changing the URL param based on if the user is viewing the form details or the summary, so it sends the user back to the `?step=details` page from `?step=summary` and throws the prompt. If they click "no" they won't be able to click "Go to Summary" since it the route won't be able to change back to summary (since it never changed back to details). Should we remove the prompt?